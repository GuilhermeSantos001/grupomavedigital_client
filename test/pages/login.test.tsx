import '@testing-library/jest-dom/extend-expect'

import React from 'react'
import { render } from '../testUtils'
import Login from '../../pages/auth/login'

describe('Login page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Login />, {})
    expect(asFragment()).toMatchSnapshot()
  })

  it('Verifica se a div principal está com as propriedades corretas', () => {
    const { getByTestId } = render(<Login />, {})

    expect(getByTestId('div-container')).toHaveClass('p-2')
    expect(getByTestId('div-container')).toHaveStyle({
      fontFamily: 'Fira Code',
    })
  })

  it('Verifica se o texto principal está correto', () => {
    const { getByTestId } = render(<Login />, {})

    expect(getByTestId('text-start')).toHaveClass('fw-bold')
    expect(getByTestId('div-container')).toHaveTextContent(
      'Conecte-se ao Ambiente Digital Interativo'
    )
  })

  it('Verifica se o separador da primeira linha está presente', () => {
    const { getByTestId } = render(<Login />, {})

    expect(getByTestId('separator')).toHaveClass('text-muted')
  })

  it('Verifica se o container do input login está correto', () => {
    const { getByTestId } = render(<Login />, {})

    expect(getByTestId('container-input-login')).toHaveClass('col-12 p-2')

    expect(getByTestId('input-container-username')).toHaveClass(
      'input-group mb-3'
    )
    expect(getByTestId('span-username')).toHaveClass('input-group-text')
    expect(getByTestId('span-username').id).toBe('username-addon')
    expect(getByTestId('input-username')).toHaveClass('form-control')
    expect(getByTestId('input-username')).toHaveProperty(
      'placeholder',
      'Nome de usuário'
    )
    expect(getByTestId('input-username')).toHaveProperty('type', 'text')

    expect(getByTestId('container-input-password')).toHaveClass('input-group')
    expect(getByTestId('span-password')).toHaveClass('input-group-text')
    expect(getByTestId('span-password').id).toBe('password-addon')
    expect(getByTestId('span-password-eye')).toHaveClass(
      'input-group-text animation-delay hover-color'
    )
    expect(getByTestId('span-password-eye').id).toBe('password-addon-eye')
    expect(getByTestId('input-password')).toHaveClass('form-control')
    expect(getByTestId('input-password')).toHaveProperty(
      'placeholder',
      'Senha de usuário'
    )
    expect(getByTestId('input-password')).toHaveProperty('type', 'password')
    expect(getByTestId('help-password')).toHaveTextContent(
      'Nunca mostre sua senha a ninguém e nem passe por e-mail.'
    )
  })

  it('Verifica se o container do button login está correto', () => {
    const { getByTestId } = render(<Login />, {})

    expect(getByTestId('container-button-login')).toHaveClass('d-grid gap-2')
    expect(getByTestId('button-login')).toHaveClass('btn btn-primary')
    expect(getByTestId('button-login')).toHaveProperty('type', 'button')
    expect(getByTestId('button-login')).toHaveTextContent('Entrar')
  })
})
