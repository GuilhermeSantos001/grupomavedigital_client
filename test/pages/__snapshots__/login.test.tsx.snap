// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="p-2"
    data-testid="div-container"
    style="font-family: Fira Code;"
  >
    <h1
      class="fw-bold"
      data-testid="text-start"
    >
      Conecte-se ao Ambiente Digital Interativo
    </h1>
    <hr
      class="text-muted"
      data-testid="separator"
    />
    <div
      class="col-12 p-2"
      data-testid="container-input-login"
    >
      <div
        class=""
        style="position: fixed; top: 0px; left: 0px; height: 2px; background: transparent; z-index: 99999999999; width: 100%;"
      >
        <div
          style="height: 100%; background: rgb(246, 216, 22); transition: all 500ms ease; width: 0%;"
        >
          <div
            style="box-shadow: 0 0 10px #f6d816, 0 0 5px #f6d816; width: 5%; opacity: 1; position: absolute; height: 100%; transition: all 500ms ease; transform: rotate(3deg) translate(0px, -4px); left: -10rem;"
          />
        </div>
      </div>
      <div
        class="input-group mb-3"
        data-testid="input-container-username"
      >
        <span
          class="input-group-text"
          data-testid="span-username"
          id="username-addon"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-user fa-w-14 fs-6 flex-shrink-1 text-primary my-auto"
            data-icon="user"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
              fill="currentColor"
            />
          </svg>
        </span>
        <input
          aria-describedby="username-addon"
          aria-label="Username"
          class="form-control"
          data-testid="input-username"
          name="username"
          placeholder="Nome de usuário"
          type="text"
          value=""
        />
      </div>
      <div
        class="input-group"
        data-testid="container-input-password"
      >
        <span
          class="input-group-text"
          data-testid="span-password"
          id="password-addon"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-key fa-w-16 fs-6 flex-shrink-1 text-primary my-auto"
            data-icon="key"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M512 176.001C512 273.203 433.202 352 336 352c-11.22 0-22.19-1.062-32.827-3.069l-24.012 27.014A23.999 23.999 0 0 1 261.223 384H224v40c0 13.255-10.745 24-24 24h-40v40c0 13.255-10.745 24-24 24H24c-13.255 0-24-10.745-24-24v-78.059c0-6.365 2.529-12.47 7.029-16.971l161.802-161.802C163.108 213.814 160 195.271 160 176 160 78.798 238.797.001 335.999 0 433.488-.001 512 78.511 512 176.001zM336 128c0 26.51 21.49 48 48 48s48-21.49 48-48-21.49-48-48-48-48 21.49-48 48z"
              fill="currentColor"
            />
          </svg>
        </span>
        <input
          aria-describedby="password-addon"
          aria-label="Password"
          class="form-control"
          data-testid="input-password"
          name="password"
          placeholder="Senha de usuário"
          type="password"
          value=""
        />
        <span
          class="input-group-text animation-delay hover-color"
          data-testid="span-password-eye"
          id="password-addon-eye"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-low-vision fa-w-18 fs-6 flex-shrink-1 my-auto"
            data-icon="low-vision"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M569.344 231.631C512.96 135.949 407.81 72 288 72c-28.468 0-56.102 3.619-82.451 10.409L152.778 10.24c-7.601-10.858-22.564-13.5-33.423-5.9l-13.114 9.178c-10.86 7.601-13.502 22.566-5.9 33.426l43.131 58.395C89.449 131.73 40.228 174.683 6.682 231.581c-.01.017-.023.033-.034.05-8.765 14.875-8.964 33.528 0 48.739 38.5 65.332 99.742 115.862 172.859 141.349L55.316 244.302A272.194 272.194 0 0 1 83.61 208.39l119.4 170.58h.01l40.63 58.04a330.055 330.055 0 0 0 78.94 1.17l-189.98-271.4a277.628 277.628 0 0 1 38.777-21.563l251.836 356.544c7.601 10.858 22.564 13.499 33.423 5.9l13.114-9.178c10.86-7.601 13.502-22.567 5.9-33.426l-43.12-58.377-.007-.009c57.161-27.978 104.835-72.04 136.81-126.301a47.938 47.938 0 0 0 .001-48.739zM390.026 345.94l-19.066-27.23c24.682-32.567 27.711-76.353 8.8-111.68v.03c0 23.65-19.17 42.82-42.82 42.82-23.828 0-42.82-19.349-42.82-42.82 0-23.65 19.17-42.82 42.82-42.82h.03c-24.75-13.249-53.522-15.643-79.51-7.68l-19.068-27.237C253.758 123.306 270.488 120 288 120c75.162 0 136 60.826 136 136 0 34.504-12.833 65.975-33.974 89.94z"
              fill="currentColor"
            />
          </svg>
        </span>
      </div>
      <div
        class="col-12 bd-callout bd-callout-primary my-2"
      >
        <p>
          <small
            data-testid="help-password"
          >
            Nunca mostre sua senha a ninguém e nem passe por e-mail.
          </small>
        </p>
      </div>
      <div
        class="d-grid gap-2"
        data-testid="container-button-login"
      >
        <button
          class="btn btn-primary"
          data-testid="button-login"
          disabled=""
          type="button"
        >
          Entrar
        </button>
      </div>
      <div
        class="twofactor-bg deactivate fixed-top d-flex flex-column"
      >
        <div
          class="twofactor deactivate col-12 overflow-auto"
        >
          <h1
            class="text-white text-center fs-1 fw-bold m-5"
          >
            Autenticação de dois fatores está ativada. Por gentileza, insira seu código de segurança.
          </h1>
          <br />
          <div
            class="col-10 mx-auto mb-5"
          >
            <label
              for="twofactor-usertoken"
            >
              Insira o código de proteção, exibido em seu gerenciador de códigos para autenticação em duas etapas.
            </label>
            <input
              class="form-control mb-2"
              type="tel"
              value=""
            />
            <button
              class="btn btn-primary col-12 my-1"
              disabled=""
              type="button"
            >
              Validar
            </button>
            <button
              class="btn btn-primary col-12 mb-1"
              type="button"
            >
              Voltar
            </button>
            <button
              class="btn btn-danger col-12 mb-2"
              type="button"
            >
              Recuperar a conta
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
